engine/apps/alerts/tasks/check_escalation_finished.py

```diff
@@ -1,48 +1,115 @@
+        requests.get(heartbeat_url).raise_for_status()
+        task_logger.info(f"Heartbeat successfully sent to {heartbeat_url}")
+    else:
+        task_logger.info(f"Skipping sending heartbeat as no heartbeat URL is configured")
+
+
+def audit_alert_group_escalation(alert_group: "AlertGroup") -> None:
+    AlertGroupLogRecord = apps.get_model("alerts", "AlertGroupLogRecord")
+
+    escalation_snapshot = alert_group.escalation_snapshot
+    alert_group_id = alert_group.id
+    base_msg = f"Alert group {alert_group_id}"
+
+    if not escalation_snapshot:
+        raise AlertGroupEscalationPolicyExecutionAuditException(
+            f"{base_msg} does not have an escalation snapshot associated with it, this should never occur"
+        )
+    task_logger.info(f"{base_msg} has an escalation snapshot associated with it, auditing if it executed properly")
+
+    escalation_policies_snapshots = escalation_snapshot.escalation_policies_snapshots
+
+    if not escalation_policies_snapshots:
+        # TODO: in what case is this valid?
+        task_logger.info(
+            f"{base_msg}'s escalation snapshot has an empty escalation_policies_snapshots, skipping further validation"
+        )
+        return
+    task_logger.info(
+        f"{base_msg}'s escalation snapshot has a populated escalation_policies_snapshots, continuing validation"
+    )
+
+    if escalation_snapshot.next_step_eta_is_valid() is False:
+        # TODO: is a next_step_eta value of None actually valid here, or should we raise an error?
+        raise AlertGroupEscalationPolicyExecutionAuditException(
+            f"{base_msg}'s escalation snapshot does not have a valid next_step_eta: {escalation_snapshot.next_step_eta}"
+        )
+    task_logger.info(f"{base_msg}'s escalation snapshot has a valid next_step_eta: {escalation_snapshot.next_step_eta}")
+
+    last_active_escalation_policy_snapshot = escalation_snapshot.last_active_escalation_policy_snapshot
+
+    if last_active_escalation_policy_snapshot is None or not last_active_escalation_policy_snapshot.escalation_policy:
+        # TODO: in what case is this valid?
+        task_logger.info(
+            f"{base_msg}'s escalation snapshot does not have a last_active_escalation_policy_snapshot, skipping further validation"
+        )
+        return
+    task_logger.info(f"{base_msg}'s escalation snapshot has a last_active_escalation_policy_snapshot")
+
+    # TODO: I'm assuming these are the proper alert group log record types to check for but want to verify
+    if not last_active_escalation_policy_snapshot.escalation_policy.log_records.filter(
+        type__in=[AlertGroupLogRecord.TYPE_ESCALATION_TRIGGERED, AlertGroupLogRecord.TYPE_ESCALATION_FINISHED]
+    ).exists():
+        # TODO: do we actually need to go through all of the escalation policy snapshots and validate log records
+        # for all of those, or is the last_active one sufficient?
```

Comment:
@iskhakov had a few minor questions for you within this block, when you have time